deploy/setup/install-api.sh

```diff
@@ -13,35 +13,36 @@ OPENTREE_DOCSTORE=$2
 COLLECTIONS_REPO=$3
 AMENDMENTS_REPO=$4
 FAVORITES_REPO=$5
-CONTROLLER=$6
+export CONTROLLER=$6
 OTI_BASE_URL=$7
 OPENTREE_API_BASE_URL=$8
 COLLECTIONS_API_BASE_URL=$9
 AMENDMENTS_API_BASE_URL=${10}
 FAVORITES_API_BASE_URL=${11}
 OPENTREE_DEFAULT_APPLICATION=${12}
 OTINDEX_BASE_URL=${13}
+READ_ONLY_MODE=${14}
 
-. setup/functions.sh
+. setup/functions.sh || exit 1
 
-setup/install-common.sh $OPENTREE_DEFAULT_APPLICATION $CONTROLLER
 
-echo "Installing API"
+bash setup/install-web2py.sh || exit 1
+
+echo "Installing API" || exit 1
 
 
 #Pre installs for normal phylestem-api venv
-venv/bin/pip  install vine==1.3
+venv/bin/pip  install vine==1.3 || exit 1
 #Force early version of vine in venv to deal with python2 python3 issues
-venv/bin/pip install kombu==4.1.0
-venv/bin/pip install redis==2.10.6
-venv/bin/pip install celery==4.1.0
+venv/bin/pip install kombu==4.1.0 || exit 1
+venv/bin/pip install celery==4.1.0 || exit 1
 
 
 
 
 # ---------- Redis for caching ---------
 #This is in the default venv
-REDIS_WITH_VERSION="redis-2.10.6"
+REDIS_WITH_VERSION="redis-3.0.0"
```

Comment:
I made this change, but based on experiments on devphylesystemapi needs to be reverted to redis==2.10.6. I don't recollect why I changed it!

Reply:
So are we running _two_ versions of redis, one inside and one outside of the python virtual environment?

Reply:
A good q.. and one that I don't honestly understand the answer to! 
on dev phylsesystem-api redis-cli 3.0.0 is installed from releases, but we were running into https://github.com/celery/celery/issues/5175  until @mtholder ran  ```venv/bin/pip install redis==2.10.6```  
But AFAIK redis 3.0.0 is still getting called by https://github.com/OpenTreeOfLife/germinator/blob/migrate-oauth-to-github-app/deploy/setup/install-api.sh#L264  I may be wrong about that part though! Mark might be able to clarify.

Reply:
I'm not sure what is running. Here is what I suspect: the redis server can probably be running 2.10.6 or 3.0.0. 
However, celery has be running through python2.7 (because we are using web2py). It imports the redis python package to interact with the redis server. I think that the means that we need to be using a not-the-latest version of the redis python package.

Reply:
ah right - that sounds correct to me!